api/src/main/java/org/geysermc/geyser/api/item/custom/v2/component/java/ItemDataComponents.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2025 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.item.custom.v2.component.java;
+
+import org.geysermc.geyser.api.GeyserApi;
+import org.geysermc.geyser.api.item.custom.v2.CustomItemDefinition;
+import org.geysermc.geyser.api.item.custom.v2.component.DataComponent;
+import org.geysermc.geyser.api.item.custom.v2.component.geyser.GeyserDataComponent;
+import org.geysermc.geyser.api.util.Identifier;
+
+import java.util.function.Predicate;
+
+/**
+ * Represents various Java item data components to the extent
+ * that these can be translated to custom items for Bedrock edition players.
+ * These can be set in {@link CustomItemDefinition.Builder#component(DataComponent, Object)}
+ * to specify the item's behavior. It's expected that these components would also
+ * be present on the item server-side to avoid de-syncs.
+ *
+ * @see CustomItemDefinition#components()
+ * @see GeyserDataComponent
+ */
+public class ItemDataComponents {
+
+    /**
+     * Marks the item as consumable. Of this component, only {@code consume_seconds} and {@code animation} properties are translated. Consume effects are done server side,
+     * and consume sounds and particles aren't possible.
```

Comment:
```suggestion
     * and consume sounds and disabling consume particles aren't possible.
```

Bedrock does display item consume particles as far as I know, but unlike on Java edition, it's not possible to disable/hide these consume particles through item components.